### PR TITLE
Support voting through DesignatedVoting from voter dapp

### PR DESCRIPTION
**Pull request by ptare**
In this first-pass implementation, if the account selected in Metamask
has a DesignatedVoting contract deployed, the voter dapp will prefer to
use that contract instead.

Follow-up PRs can improve the UX (significantly) and build a flow
to deploy the contract from the dapp.

Signed-off-by: Prasad Tare <prasad.tare@gmail.com>

### CODE REVIEW

File: voter-dapp/src/drizzleOptions.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
+import DesignatedVotingFactory from "./contracts/DesignatedVotingFactory.json";
 import Voting from "./contracts/Voting.json";
 import VotingToken from "./contracts/VotingToken.json";
 
 const options = {
-  contracts: [Voting, VotingToken],
+  contracts: [DesignatedVotingFactory, Voting, VotingToken],
```
Comment (mrice32) [inline voter-dapp/src/drizzleOptions.js]: nit: are we using the DesignatedVotingFactory?

File: core/contracts/DesignatedVoting.sol [hunk 2]
```diff
@@ -42,13 +42,27 @@ contract DesignatedVoting is MultiRole, Withdrawable {
         _getVotingAddress().commitVote(identifier, time, hash);
     }
 
+    /**
+     * @notice Forwards a batch commit to Voting.
+     */
+    function batchCommit(Voting.Commitment[] calldata commits) external {
```
[the remark this answers is not in the document]
Comment (ptare) [reply core/contracts/DesignatedVoting.sol]: Yep, thanks :)

File: voter-dapp/src/drizzleOptions.js (hunk 1 above)
Comment (ptare) [reply voter-dapp/src/drizzleOptions.js]: We use it in the Dashboard component to look up the voter's DesignatedVoting contract.

File: voter-dapp/src/ActiveRequests.js [hunk 3]
```diff
@@ -195,7 +197,7 @@ function ActiveRequests() {
     if (commits.length < 1) {
       return;
     }
-    batchCommitFunction(commits);
+    batchCommitFunction(commits, { from: account });
```
[the remark this answers is not in the document]
Comment (ptare) [reply voter-dapp/src/ActiveRequests.js]: To be honest, I don't know when/why you need that. I added it to reveal just to be a bit more consistent.

File: voter-dapp/src/drizzleOptions.js (hunk 1 above)
Comment (mrice32) [reply voter-dapp/src/drizzleOptions.js]: Oh right, of course. Thanks.

**Comment by ptare** [comment]
I added and then removed a commit from this branch. It's now back to commit d3178cf, which is the last commit you looked at.

I realized that this PR is incomplete, in that it doesn't filter events properly, which I started to fix before recognizing that fixing events would be a larger change than I had thought. I'll fix those in a follow up PR.

**Comment by ptare** [comment]
Issue #787

**Pull request merged**